File: src/BreakfastProvider.Api/Models/Requests/StaffMemberRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record StaffMemberRequest
{
    public string? Name { get; init; }
    public string? Role { get; init; }
    public string? Email { get; init; }
    public bool IsActive { get; init; } = true;
    public DateTime? HiredAt { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/StaffMemberRequest.java
package io.lemonlion.breakfast.model.request;

import java.time.Instant;

/** Twin of C# {@code StaffMemberRequest}. {@code isActive} defaults to true when omitted. */
public record StaffMemberRequest(String name, String role, String email, Boolean isActive, Instant hiredAt) {

    public boolean activeOrDefault() {
        return isActive == null || isActive;
    }
}
